be/src/vec/sink/vdata_stream_sender.cpp

```diff
@@ -138,13 +144,27 @@ Status VDataStreamSender::Channel::send_block(PBlock* block, bool eos) {
     _closure->ref();
     _closure->cntl.set_timeout_ms(_brpc_timeout_ms);
 
-    if (_brpc_request.has_block()) {
-        request_block_transfer_attachment<PTransmitDataParams,
-                                          RefCountClosure<PTransmitDataResult>>(
+    if (_parent->_column_values_buffer_ptr != nullptr && _brpc_request.has_block() &&
+        !_brpc_request.block().has_column_values()) {
+        DCHECK(_parent->_column_values_buffer.size() != 0);
+        request_embed_attachment_contain_block<PTransmitDataParams,
+                                               RefCountClosure<PTransmitDataResult>>(
                 &_brpc_request, _parent->_column_values_buffer, _closure);
+        std::string brpc_url;
+        brpc_url =
```

Comment:
this `brpc_url` and `_brpc_http_stub` can be initialized when creating data_stream_sender.

Reply:
`brpc_url` and `_brpc_http_stub` are only used in rare cases where tuple/block > 2G, and no initialization is required in most cases.